### PR TITLE
Add snippets for php-mode

### PR DESCRIPTION
**Pull request by zonuexe**
Add magic methods, popular [PHPDoc tags](https://phpdoc.org/docs/latest/internals/tags.html) and convenience idioms for debugging.

### CODE REVIEW

**Comment by AndreaCrotti** [comment]
Thanks a lot @zonuexe I'll try to have a deeper look next week!

File: snippets/php-mode/license-doc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- mode: snippet -*-
+# contributor: USAMI Kenta <tadsan@zonu.me>
+# name: PHPDoc @license
+# key: *license
```
Comment (AndreaCrotti) [inline snippets/php-mode/license-doc]: Any reason to prepend all license snippets with *?
It doesn't make it easy to remember the snippet imho..

**Comment by AndreaCrotti** [comment]
I'm not sure about all the keys prefixed by "*", also because I don't see a pattern for which ones should be and which ones are not prefixed.

If we can clarify/fix that @zonuexe then I'm happy to merge.

**Comment by zonuexe** [comment]
@AndreaCrotti 
It can be explained by the place where @AndreaCrotti license is written.
Snippets with "Doc" in their name are carefully designed based on the actual input process.

## Manual typing

yasnippet-php-freehand

## Use yasnippet

yasnippet-php-yasnippet

------

`@license` has to be written in DocComment, so it is always entered with `* @`.
I believe that `*` is reasonable and best because of the connectivity of the input with other tags for the PHPDoc tag prefix.

**Comment by zonuexe** [comment]
I found my typo with this video ;-)

**Comment by AndreaCrotti** [comment]
Yes ok thanks @zonuexe makes sense

**Comment by zonuexe** [comment]
Thanks!

**Pull request merged**